### DISCUSSION TITLE
Can't upload files over 1MB, v 4.12.0

### DISCUSSION THREAD

**Discussion by hess8** [LycheeOrg/Lychee#2022] [Q&A] [answered]
I migrated my Lychee v4 database and files to a new server, with the latest Lychee code and can now only upload tiny files, less than about 1 MB!   The error says that the "post_max_size may be too small".

The logs show no errors.

But I made the post_max_size very large.  The Diagnostics page shows the recommended php settings:
    Max uploaded file size:                  100M
    Max post size:                                100M
    Max execution time:                       200

Any ideas of why Lychee isn't using the max sizes I've set?

_______________________________________________________________________________

Diagnostics page:

Diagnostics
    -------
    Info: Latest version of PHP is 8.2
    Foreign key: access_permissions.user_id     → users.id             : CASCADE
    Foreign key: albums.id                      → base_albums.id       : NO ACTION
    Foreign key: base_albums.owner_id           → users.id             : NO ACTION
    Foreign key: jobs_history.owner_id          → users.id             : NO ACTION
    Foreign key: photos.album_id                → albums.id            : NO ACTION
    Foreign key: photos.owner_id                → users.id             : NO ACTION
    Foreign key: size_variants.photo_id         → photos.id            : NO ACTION
    Foreign key: sym_links.size_variant_id      → size_variants.id     : NO ACTION
    Foreign key: tag_albums.id                  → base_albums.id       : NO ACTION

    System Information
    --------------
    Lychee Version (git):                    master (177e08c) -- Up to date (46 minutes ago).
    DB Version:                              4.12.0
    
    composer install:                        --no-dev
    APP_ENV:                                 development
    APP_DEBUG:                               false
    
    System:                                  Linux
    PHP Version:                             8.1.2-1ubuntu2.14
    PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
    Timezone:                                America/Denver
    Max uploaded file size:                  100M
    Max post size:                           100M
    Max execution time:                      200
    SQLite Version:                          3.37.2
    
    exec() Available:                        yes
    Imagick Available:                       1
    Imagick Enabled:                         1
    Imagick Version:                         1691
    GD Version:                              2.3.0
    Number of foreign key:                   9 found.

    Config Information
    --------------
    version:                                 041200
    check_for_updates:                       1
    sorting_photos_col:                      title
    sorting_photos_order:                    ASC
    sorting_albums_col:                      title
    sorting_albums_order:                    DESC
    imagick:                                 1
    skip_duplicates:                         1
    small_max_width:                         0
    small_max_height:                        360
    medium_max_width:                        1920
    medium_max_height:                       1080
    lang:                                    en
    image_overlay_type:                      desc
    default_license:                         none
    compression_quality:                     90
    grants_full_photo_access:                1
    delete_imported:                         0
    mod_frame_enabled:                       1
    mod_frame_refresh:                       30
    thumb_2x:                                1
    small_2x:                                1
    medium_2x:                               1
    landing_page_enable:                     0
    site_owner:                              
    landing_title:                           
    landing_subtitle:                        
    sm_facebook_url:                         
    sm_flickr_url:                           
    sm_twitter_url:                          
    sm_instagram_url:                        
    sm_youtube_url:                          
    landing_background:                      dist/cat.webp
    site_title:                              Lychee v4
    footer_show_copyright:                   0
    site_copyright_begin:                    1930
    site_copyright_end:                      1931
    footer_additional_text:                  
    footer_show_social_media:                0
    public_search:                           0
    SL_enable:                               0
    SL_for_admin:                            0
    recent_age:                              1
    grants_download:                         1
    photos_wraparound:                       1
    map_display:                             1
    zip64:                                   1
    map_display_public:                      0
    map_provider:                            Wikimedia
    force_32bit_ids:                         0
    map_include_subalbums:                   1
    update_check_every_days:                 100
    has_exiftool:                            0
    share_button_visible:                    1
    import_via_symlink:                      0
    has_ffmpeg:                              1
    location_decoding:                       1
    location_decoding_timeout:               30
    location_show:                           1
    location_show_public:                    0
    rss_enable:                              0
    rss_recent_days:                         7
    rss_max_items:                           100
    prefer_available_xmp_metadata:           0
    editor_enabled:                          1
    lossless_optimization:                   0
    swipe_tolerance_x:                       150
    swipe_tolerance_y:                       250
    local_takestamp_video_formats:           .avi|.mov
    log_max_num_line:                        1000
    unlock_password_photos_with_url_param:   0
    nsfw_visible:                            1
    nsfw_blur:                               0
    nsfw_warning:                            0
    nsfw_warning_admin:                      0
    nsfw_banner_override:                    
    map_display_direction:                   1
    album_subtitle_type:                     oldstyle
    upload_processing_limit:                 4
    public_photos_hidden:                    0
    new_photos_notification:                 0
    legacy_id_redirection:                   1
    zip_deflate_level:                       6
    SA_enabled:                              1
    default_album_protection:                1
    album_decoration:                        layers
    album_decoration_orientation:            row
    allow_username_change:                   1
    auto_fix_orientation:                    1
    use_job_queues:                          0
    random_album_id:                         starred
    use_last_modified_date_when_no_exif_date: 0
    ffmpeg_path:                             /usr/bin/ffmpeg
    ffprobe_path:                            /usr/bin/ffprobe
    layout:                                  justified

**Comment by d7415** [comment]
Check the settings of the httpd and any reverse proxies between you and it.

**Comment by ildyria** [comment]
> I migrated my Lychee v4 database and files to a new server, with the latest Lychee code and can now only upload tiny files, less than about 1 MB! The error says that the "post_max_size may be too small".

The good news is this will be fixed in the next version.
V5 (#1303) give the ability to upload by chunk. :)

**Comment by hess8** [comment]
httpd is for Apache, right?  I'm using Nginx

```
    location = /index.php {
        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
        fastcgi_param HTTP_PROXY "";
        fastcgi_pass unix:/run/php/php8.1-fpm.sock;
        fastcgi_index index.php;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param PHP_VALUE 
"max_execution_time=200
 post_max_size=100M
            upload_max_size = 100M
            upload_max_filesize=100M
            memory_limit=256M";
        fastcgi_param PATH /usr/local/bin:/usr/bin:/bin;
        include fastcgi_params;
    }
```

**Reply by d7415** [reply, marked as answer]
httpd is generic. Check your `client_max_body_size` in nginx
https://lycheeorg.github.io/docs/faq_troubleshooting.html#i-cant-upload-large-photos

**Reply by hess8** [reply]
Yes, that was the problem.  Thanks!
